pkg/apis/pipeline/v1beta1/pipeline_validation.go

```diff
@@ -523,3 +523,48 @@ func validateResultsFromMatrixedPipelineTasksNotConsumed(tasks []PipelineTask, f
 	}
 	return errs
 }
+
+// ValidateParamArrayIndex validates if the param reference to an array param is out of bound.
+// error is returned when the array indexing reference is out of bound of the array param
+// e.g. if a param reference of $(params.array-param[2]) and the array param is of length 2.
+func (pipelineSpec *PipelineSpec) ValidateParamArrayIndex(ctx context.Context, params []Param) error {
```

Comment:
linting is failing with:

```
pkg/apis/pipeline/v1beta1/pipeline_validation.go:530:1: receiver-naming: receiver name pipelineSpec should be consistent with previous receiver name ps for PipelineSpec (revive)
```

will have to change this to:

```
func (ps *PipelineSpec) ValidateParamArrayIndex(ctx context.Context, params []Param) error {
```

We have another member function of `pipelineSpec` with the receiver - `ps` at https://github.com/tektoncd/pipeline/blob/main/pkg/apis/pipeline/v1beta1/pipeline_validation.go#L54

Reply:
oh thanks!